nucypher/network/server.py

```diff
@@ -74,157 +79,109 @@ def rest_url(self):
-                  'DELETE',
-                  self.revoke_arrangement),
-            Route('/public_information', 'GET',
-                  self.public_information),
-            Route('/node_metadata', 'GET',
-                  self.all_known_nodes),
-            Route('/node_metadata', 'POST',
-                  self.node_metadata_exchange),
-            Route('/consider_arrangement',
-                  'POST',
-                  self.consider_arrangement),
-            Route('/treasure_map/{treasure_map_id}',
-                  'GET',
-                  self.provide_treasure_map),
-            Route('/status',
-                  'GET',
-                  self.status),
-            Route('/treasure_map/{treasure_map_id}',
-                  'POST',
-                  self.receive_treasure_map),
-        ]
-
-        self.rest_app = App(routes=routes)
-        self.db_filepath = db_filepath
-
-        from nucypher.keystore import keystore
-        from nucypher.keystore.db import Base
-        from sqlalchemy.engine import create_engine
-
-        self.log.info("Starting datastore {}".format(self.db_filepath))
-
-        # See: https://docs.sqlalchemy.org/en/rel_0_9/dialects/sqlite.html#connect-strings
-        if self.db_filepath:
-            db_uri = f'sqlite:///{self.db_filepath}'
-        else:
-            db_uri = 'sqlite://'   # TODO: Is this a sane default? See #667
-
-        engine = create_engine(db_uri)
-
-        Base.metadata.create_all(engine)
-        self.datastore = keystore.KeyStore(engine)
-        self.db_engine = engine
-
-        from nucypher.characters.lawful import Alice, Ursula
-        self._alice_class = Alice
-        self._node_class = Ursula
-
-        with open(os.path.join(TEMPLATES_DIR, "basic_status.j2"), "r") as f:
-            _status_template_content = f.read()
-        self._status_template = Template(_status_template_content)
-
-    def public_information(self):
+def make_rest_app(
+        db_filepath: str,
+        network_middleware: RestMiddleware,
+        federated_only: bool,
+        treasure_map_tracker: dict,
+        node_tracker: 'FleetStateTracker',
+        node_bytes_caster: Callable,
+        work_order_tracker: list,
+        node_recorder: Callable,
+        stamp: SignatureStamp,
+        verifier: Callable,
+        suspicious_activity_tracker: dict,
+        serving_domains,
+        log=Logger("http-application-layer")
+) -> Tuple:
+    forgetful_node_storage = ForgetfulNodeStorage(federated_only=federated_only)
+
+    from nucypher.keystore import keystore
+    from nucypher.keystore.db import Base
+    from sqlalchemy.engine import create_engine
+
+    log.info("Starting datastore {}".format(db_filepath))
+
+    # See: https://docs.sqlalchemy.org/en/rel_0_9/dialects/sqlite.html#connect-strings
+    if db_filepath:
+        db_uri = f'sqlite:///{db_filepath}'
+    else:
+        db_uri = 'sqlite://'  # TODO: Is this a sane default? See #667
```

Comment:
Should this not be `:memory:` or whatever the SQLAlchemy/sqlite thing is?

Reply:
Not... sure... that's part of a rebase of @KPrasch's.